prawns/balances/src/handlers/typeGetters/getTreasuryEvents.ts

```diff
@@ -46,3 +55,66 @@ export function getTreasuryDepositEvent(
     }
   }
 }
+
+
+export function getTreasuryAwardedEvent(
+  ctx: EventHandlerContext,
+  network: SubstrateNetwork
+): {
+  award: bigint;
+  account: Uint8Array;
+} {
+  switch (network) {
+    case SubstrateNetwork.phala: {
+      const event = new KhalaTreasuryAwardedEvent(ctx);
+
+      if (event.isV1) {
+        const [, award, account] = event.asV1;
+        return {
+          award,
+          account
+        };
+      } else if (event.isV1110) {
+        return event.asV1110;
+      } else {
+        return event.asLatest;
+      }
+    }
+
+    case SubstrateNetwork.polkadot: {
+      const event = new PolkadotTreasuryAwardedEvent(ctx);
+
+      if (event.isV0) {
+        const [, award, account] = event.asV0;
+        return {
+          award,
+          account
+        };
+      } else if (event.isV9170) {
+        return event.asV9170;
+      } else {
+        return event.asLatest;
+      }
+    }
+
+    case SubstrateNetwork.kusama: {
+      const event = new KudamaTreasuryAwardedEvent(ctx);
+
+      if (event.isV1020) {
+        const [, award, account] = event.asV1020;
+        return {
+          award,
+          account
+        };
+      } else if (event.isV9160) {
+        return event.asV9160;
+      } else {
+        return event.asLatest;
+      }
+    }
+
+    default: {
+      throw new Error('getBalancesWithdrawEvent::network not supported');
```

Comment:
It should be:
`getTreasuryAwardedEvent::network not supported`

😄

Reply:
eagle eyes